src/Loaders/InsertUpdate.php

```diff
@@ -249,7 +256,9 @@ protected function execute($row): void
 
         $current = $this->select->fetch();
         if (false !== $current) {
-            $this->update($row, $current);
+            if ($this->doUpdates) {
```

Comment:
Hello, thanks for the PR :)
Maybe we can do this in one statement instead of nested conditions ?
```suggestion
         if (
             false !== $current
             && true === $this->doUpdates
         ) {
```

Reply:
Ummm - no. That is a different logic. That would cause the row to be inserted rather than skipped.

I pushed an additional check in the test code so it you want you can try the change on a local checkout and see why it fails.

Reply:
oh right, got it. I have some suggestion to make this more intuitive

only check for `$current` here
```php
        ...
        $current = $this->select->fetch();

        if (false === $current) {
            $this->insert($row);
        } else {
            $this->update($row, $current);
        }
```
and do the skip in the `update` method
```php
    protected function update(array $row, array $current): void
    {
        if (false === $this->doUpdates) {
            return;
        }
        ...
```
What do you think ?

Reply:
I think that would work (and the modified test should help verify)

It does mean that your fork will be more out of sync with the original work, but it seems like that is not a concern for you. I'll try the change out.

Reply:
@kdebisschop yes our fork it totally decoupled from the original work since this one has too much drawbacks (for instance the hard coupling with Laravel).